bot/cogs/queue.py

```diff
@@ -344,11 +282,13 @@ async def cap(self, ctx, *args):
             except ValueError:
                 embed = discord.Embed(title=f'{new_cap} is not an integer', color=self.bot.color)
             else:
-                if new_cap < 2 or new_cap > 100:
+                if new_cap == capacity:
+                    embed = discord.Embed(title=f'Capacity is already set to {capacity}', color=self.bot.color)
+                elif new_cap < 2 or new_cap > 100:
                     embed = discord.Embed(title='Capacity is outside of valid range', color=self.bot.color)
                 else:
-                    queue.active.clear()  # Empty active queue to prevent bugs related to capacity size
-                    queue.capacity = new_cap
+                    await self.bot.db_helper.delete_all_queued_users(ctx.guild)
+                    await self.bot.db_helper.update_guild(ctx.guild, capacity=new_cap)
```

Comment:
Could you also run these in parallel or do they need to be called in this order?

Reply:
It's probably better to make sure they are called in that order. We don't ever want the queue in a state where the queue size is larger than the capacity.

Reply:
Yeah that makes sense, just want to make sure it's as efficient as possible :P